src/Ttl.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Cache;
+
+use DateInterval;
+use DateTime;
+
+/**
+ * Value object representing a time-to-live (TTL) duration in seconds.
+ *
+ * ```php
+ * $ttl = Ttl::minutes(5);
+ * $seconds = $ttl->toSeconds(); // 300
+ * ```
+ */
+final class Ttl
+{
+    private const SECONDS_IN_MINUTE = 60;
+    private const SECONDS_IN_HOUR = 3600;
+    private const SECONDS_IN_DAY = 86400;
+
+    private const INT_PLACEHOLDER_FOR_FOREVER_VALUE = 0;
+
+    private function __construct(
+        public readonly int $value,
+        public readonly bool $isForever = false
```

Comment:
```suggestion
        public readonly int|null $value,
```

Null means TTL is forever.

May be all negative values normalize to zero?